src/Abstractions/TaskOptions.cs

```diff
@@ -6,153 +6,61 @@ namespace Microsoft.DurableTask;
-        /// Configures a task retry policy.
-        /// </summary>
-        /// <param name="policy">The task retry policy to configure.</param>
-        /// <returns>Returns the current <see cref="Builder"/> object.</returns>
-        public Builder WithRetryStrategy(RetryPolicy policy)
-        {
-            this.RetryPolicy = policy;
-            this.RetryHandler = null;
-            return this;
-        }
-
-        /// <inheritdoc cref="WithRetryStrategy(AsyncRetryHandler)"/>
-        public Builder WithRetryStrategy(RetryHandler handler)
-        {
-            // Synchronous handlers are wrapped in an async handler so that we only have
-            // to keep track of a single handler assignment.
-            return this.WithRetryStrategy(retryContext => Task.FromResult(handler(retryContext)));
-        }
-
-        /// <summary>
-        /// Configures a retry handler.
-        /// </summary>
-        /// <param name="handler">The handler to invoke when deciding whether to retry a failed orchestrator task.</param>
-        /// <returns>Returns the current <see cref="Builder"/> object.</returns>
-        public Builder WithRetryStrategy(AsyncRetryHandler handler)
-        {
-            this.RetryHandler = handler;
-            this.RetryPolicy = null;
-            return this;
-        }
-
-        /// <summary>
-        /// Configures a <see cref="CancellationToken"/> that can be used to cancel the task execution.
-        /// </summary>
-        /// <remarks>
-        /// Cancellation tokens can be used to stop the current orchestrator from awaiting a pending activity or
-        /// sub-orchestration completion. However, this cancellation won't necessarily stop the activity or
-        /// sub-orchestration from running in the background.
-        /// </remarks>
-        /// <param name="cancellationToken">The cancellation token to use for cancelling task execution.</param>
-        /// <returns>Returns the current <see cref="Builder"/> object.</returns>
-        public Builder WithCancellationToken(CancellationToken cancellationToken)
-        {
-            if (cancellationToken != default)
-            {
-                throw new NotSupportedException(
-                    "Durable task cancellation is not yet supported. See"
-                    + " https://github.com/microsoft/durabletask-dotnet/issues/7 for more information.");
-            }
-
-            this.CancellationToken = cancellationToken;
-            return this;
-        }
+/// <summary>
+/// Options that can be used to control the behavior of orchestrator task execution. This derived type can be used to
+/// supply extra options for orchestrations.
+/// </summary>
+/// <param name="InstanceId">The orchestration instance ID to use. <c>null</c> to have one generated.</param>
+/// <param name="Retry">The retry options. <c>null</c> for no retries.</param>
+public record OrchestrationOptions(string? InstanceId = null, TaskRetryOptions? Retry = null)
```

Comment:
Open to suggestions for a better name (if one is needed that is). Maybe `SubOrchestrationOptions` because it technically only applies to sub-orchestrations?

Reply:
I was thinking the same thing about `SubOrchestrationOptions` as being a better name, mainly because I can't think of a case where we'd want to reuse this type for anything other than sub-orchestrations.